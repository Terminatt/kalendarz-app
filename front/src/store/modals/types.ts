export interface ModalState {
  isVisible: boolean;
  modalType: ModalType;
}

export enum ModalType {
  NONE,
  LOGIN_MODAL,
  REGISTER_MODAL,
}
