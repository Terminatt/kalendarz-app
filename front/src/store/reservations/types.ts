import {
    BaseDataState, BaseItem, Id,
} from '@generics/generics';
import { Room } from '@store/rooms/types';
import { User } from '@store/user/types';
import { Dayjs } from 'dayjs';

export interface ReservationState<T extends BaseItem> extends BaseDataState<T> {
    hashMapData: ReservationHashMap;
    roomReservation: Reservation[]
}

export interface ReservationHashMap {
    [key: string]: ReservationWithParsedDate[];
}

export interface BaseReservation {
    start: string;
    end: string;
    confirmed?: boolean;
    room: Id;
}

export interface UpdateReservationPayload extends Partial<BaseReservation> {
    id: Id,
    user?: Id;
}
export interface Reservation extends BaseItem, BaseReservation {
    created: string;
    user: User;
}

export interface ReservationWithUserId extends Omit<Reservation, 'user'> {
    user: Id;
}

export interface FullDataReservation extends Omit<Reservation, 'room'>, BaseItem, Omit<BaseReservation, 'room'> {
    room: Room;
    created: string;
    user: User;
}

export interface ReservationWithParsedDate extends Omit<BaseReservation, 'start' | 'end' | 'room'>, BaseItem {
    start: Dayjs;
    end: Dayjs;
    color: string;
    user: User
}
