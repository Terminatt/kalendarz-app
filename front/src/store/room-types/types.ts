import { BaseItem, ValidationErrorItem } from '@generics/generics';

export interface RoomType extends BaseItem {
    color: string;
}

export interface RoomTypeErrorResponse {
    [key: string]: ValidationErrorItem;
    name: ValidationErrorItem;
}
