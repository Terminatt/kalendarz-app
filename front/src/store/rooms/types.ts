import { BaseItem, ValidationErrorItem } from '@generics/generics';
import { RoomType } from '@store/room-types/types';

export interface BaseRoom extends BaseItem {
  floor: string;
  capacity: number;
}

export interface Room extends BaseRoom {
  type: RoomType;
  created: string;
}

export interface RoomWithTypeId extends BaseRoom {
  type: number;
}

export interface RoomErrorResponse {
  [key: string]: ValidationErrorItem;
  name: ValidationErrorItem;
}
