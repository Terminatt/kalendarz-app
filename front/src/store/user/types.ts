import {
    BaseDataState, BaseItem, Id, ValidationErrorItem,
} from '@generics/generics';

export interface UserState<T extends BaseItem> extends BaseDataState<T> {
  loadingScreen: boolean;
  currentUser: T | null
}

export interface BaseUser extends BaseItem {
  firstName: string;
  lastName: string;
  email: string;
  username: string;
  groups: Group;
  permaBanned?: boolean;
  bannedTill?: string | null;
}

export interface BaseUserUpdatePayload extends Partial<BaseItem> {
  id: Id;
}
export interface User extends BaseUser {
  created: string;
}

export interface UserErrorResponse {
  [key: string]: ValidationErrorItem;
  email: ValidationErrorItem;
  username: ValidationErrorItem;
  password: ValidationErrorItem;
}

export interface LoginErrorResponse {
  bannedTill: string;
}

export enum Group {
  NONE,
  ADMIN,
  REGULAR_USER,
}
