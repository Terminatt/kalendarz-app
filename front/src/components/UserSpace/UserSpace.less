@import '~styles/variables.less';

.signing {
  &-login {
    margin-right: @margin-medium;
    vertical-align: top;
  }

  &-register {
    display: inline-block;
  }

  &-forget-password {
    text-decoration: underline;
    color: @link-color;
    font-size: @font-size-tiny;
    text-align: center;
    margin-top: @margin-tiny;
    cursor: pointer;
  }

}