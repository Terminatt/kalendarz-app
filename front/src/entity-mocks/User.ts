import { Group, User } from '@store/user/types';
import dayjs from 'dayjs';

export const userMock: User = {
    id: 1,
    firstName: 'Test',
    lastName: 'User',
    created: dayjs().toISOString(),
    email: 'test@gmail.com',
    username: 'Test User',
    groups: Group.REGULAR_USER,
};

export const adminUserMock: User = {
    id: 2,
    firstName: 'Test',
    lastName: 'User',
    created: dayjs().toISOString(),
    email: 'test@gmail.com',
    username: 'Test User',
    groups: Group.ADMIN,
};
