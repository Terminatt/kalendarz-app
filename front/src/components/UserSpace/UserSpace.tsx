import CustomButton from '@components/CustomButton/CustomButton';
import SigningModal from '@components/Modals/SigningModal/SigningModal';
import { openModal } from '@store/modals/actions';
import { ModalType } from '@store/modals/types';
import ButtonGroup from 'antd/lib/button/button-group';
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import './UserSpace.less';

const UserSpace: React.FC = () => {
    const [isLogged, setIsLogged] = useState<boolean>(false);
    const dispatch = useDispatch();

    const openSigningModal = (modalType: ModalType) => {
        dispatch(openModal(modalType));
    };

    return isLogged ? (
        <span>Test</span>
    ) : (
        <ButtonGroup className="signing">
            <CustomButton
                className="signing-login"
                variant="clear"
                onClick={() => openSigningModal(ModalType.LOGIN_MODAL)}
            >
                Zaloguj się
            </CustomButton>
            <div className="signing-register">
                <CustomButton
                    onClick={() => openSigningModal(ModalType.REGISTER_MODAL)}
                >
                    Rejestracja
                </CustomButton>
                <div className="signing-forget-password">
                    Nie pamiętasz hasła?
                </div>
            </div>

            <SigningModal />
        </ButtonGroup>
    );
};

export default UserSpace;
