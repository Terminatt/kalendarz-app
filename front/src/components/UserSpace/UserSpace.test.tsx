import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '@store/index';
import UserSpace from './UserSpace';
import { fireEvent, render } from '@utils/testing';
import { MemoryRouter } from 'react-router';
import { ModalType } from '@store/modals/types';
import { userMock } from '@entity-mocks/User';

describe('Custom Empty Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <MemoryRouter>
                        <UserSpace />
                    </MemoryRouter>
                </Provider>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('renders buttons when user is not logged in', async () => {
        const { element } = render(
            <MemoryRouter>
                <UserSpace />
            </MemoryRouter>
        );

        expect(await element.queryByText('Zaloguj się')).not.toBeNull();
        expect(await element.queryByText('Rejestracja')).not.toBeNull();
    });


    it('triggers openSigningModal when clicking login button', async () => {
        const spy = jest.spyOn(console, 'log');

        const { element } = render(
            <MemoryRouter>
                <UserSpace />
            </MemoryRouter>
        );
        fireEvent.click(await element.findByText('Zaloguj się'));

        expect(spy).toBeCalledTimes(1);
        expect(spy).toBeCalledWith(ModalType.LOGIN_MODAL);
    });

    it('triggers openSigningModal when clicking register button', async () => {
        const spy = jest.spyOn(console, 'log');

        const { element } = render(
            <MemoryRouter>
                <UserSpace />
            </MemoryRouter>
        );
        fireEvent.click(await element.findByText('Rejestracja'));

        expect(spy).toBeCalledTimes(1);
        expect(spy).toBeCalledWith(ModalType.REGISTER_MODAL);
    });

    it('renders user name and last name when user is logged in', async () => {
        const { element } = render(
            <MemoryRouter>
                <UserSpace />
            </MemoryRouter>,
            {preloadedState: {user: {currentUser: {...userMock} } }}
        );

        expect(await element.queryByText(`${userMock.firstName} ${userMock.lastName}`)).not.toBeNull();
    });

    it('triggers onLogout', async () => {
        const spy = jest.spyOn(console, 'log');

        const { element } = render(
            <MemoryRouter>
                <UserSpace />
            </MemoryRouter>,
            {preloadedState: {user: {currentUser: {...userMock} } }}
        );

        fireEvent.click(await element.findByText('Wyloguj się'));

        expect(spy).toBeCalledTimes(1);
    });
    
});
