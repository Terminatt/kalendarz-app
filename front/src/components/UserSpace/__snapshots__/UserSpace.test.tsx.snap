// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Empty Component matches the snapshot 1`] = `
<div
  className="signing"
>
  <button
    className="ant-btn custom-btn custom-btn-clear signing-login"
    onClick={[Function]}
    type="button"
  >
    <span>
      Zaloguj siÄ™
    </span>
  </button>
  <div
    className="signing-register"
  >
    <button
      className="ant-btn custom-btn custom-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Rejestracja
      </span>
    </button>
  </div>
</div>
`;
